### PR TITLE
Fixes #188 - Show banner to users to suggest activating autofill

### LINKED ISSUE #188: Android: Show banner to users to suggest activating autofill
Most users seem to not know that AuthPass on android supports autofill.
At least for Android >= 10 we should probably display a banner in the password list, similar to other banners already implemented if the user has not yet configured AuthPass as password manager.

### PR DESCRIPTION
**Pull request by ATofighi**
Fixes #188 

I added autofillService to password_list and use a app_data to store dismiss action.

image

### CODE REVIEW

File: authpass/lib/ui/screens/password_list.dart [hunk 1]
```diff
@@ -923,11 +939,49 @@ class _PasswordListContentState extends State<PasswordListContent>
     return null;
   }
 
+  List<Widget>? _buildAutofillSuggestBanners() {
+    if (_autofillStatus == null ||
+        _autofillStatus != AutofillServiceStatus.disabled) {
+      return null;
+    }
+    if (_dismissedAutofillSuggestion != null &&
+        _dismissedAutofillSuggestion == true) {
```
Comment (hpoul) [inline authpass/lib/ui/screens/password_list.dart:948]:
```suggestion
    if (_dismissedAutofillSuggestion == true) {
```
the check for `!= null` feels a bit redundant

File: authpass/lib/ui/screens/password_list.dart [hunk 2]
```diff
@@ -363,6 +364,10 @@ class _PasswordListContentState extends State<PasswordListContent>
 
 //  final _isolateRunner = IsolateRunner.spawn();
 
+  late AutofillService _autofill;
```
Comment (hpoul) [inline authpass/lib/ui/screens/password_list.dart:367]: `_autofill` is not used outside of `_updateAutofillPrefs()` so we could probably get rid of that attribute?

File: authpass/lib/ui/screens/password_list.dart [hunk 3]
```diff
@@ -923,11 +939,49 @@ class _PasswordListContentState extends State<PasswordListContent>
...
+            child: Text(loc.dismissAutofillSuggestionBannerButton),
+          ),
+          TextButton(
+            onPressed: () {
+              _autofill.requestSetAutofillService().then((result) {
+                _autofill.status().then((value) {
+                  _autofillStatus = value;
+                });
+              });
+            },
```
Comment (hpoul) [inline authpass/lib/ui/screens/password_list.dart:971]: i think i'd prefer using async/await syntax, is a bit more consice:
```suggestion
            onPressed: () async {
              await _autofill.requestSetAutofillService();
              await _updateAutofillPrefs();
            },
```

**Review by hpoul — CHANGES_REQUESTED** [review]
Thanks again! nice work, i've added a few suggestions maybe you can take a look and see if those changes make sense to you.

**Comment by ATofighi** [comment]
> Thanks again! nice work, i've added a few suggestions maybe you can take a look and see if those changes make sense to you.

@hpoul Thank you, I'll look at them when I find free time in next days.

File: authpass/lib/ui/screens/password_list.dart (hunk 2 above)
Comment (ATofighi) [reply authpass/lib/ui/screens/password_list.dart]: fixed.

**Review by hpoul — APPROVED** [review]
Great, looks good to me 👍️ i just added a little check that only one banner is shown at a time (not both, the backup banner and the autofill banner.. we shouldn't show more than one at a time).
Anyway, great work, i'll merge it, thanks

**Pull request merged**